Sources/Diagnostics/DiagnosticsFileHandler.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  DiagnosticsFileHandler.swift
+//
+//  Created by Nacho Soto on 6/16/23.
+
+import Foundation
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+protocol DiagnosticsFileHandlerType: AnyObject {
+
+    func getEntries() async -> [DiagnosticsEvent]
+    func appendEvent(diagnosticsEvent: DiagnosticsEvent) async
+    func cleanSentDiagnostics(diagnosticsSentCount: Int) async
+    func deleteDiagnosticsFile() async
+
+}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+actor DiagnosticsFileHandler: DiagnosticsFileHandlerType {
+
+    private let fileHandler: FileHandler
+
+    init?() {
+        do {
+            self.fileHandler = try .init(Self.diagnosticsFile)
+        } catch {
+            Logger.error("Initialization error: \(error.localizedDescription)")
+            return nil
+        }
+    }
+
+    init(_ fileHandler: FileHandler) {
+        self.fileHandler = fileHandler
+    }
+
+    func appendEvent(diagnosticsEvent: DiagnosticsEvent) async {
+        guard let jsonString = diagnosticsEvent.toJSONString() else {
```

Comment:
I imagine this was probably written before we had the Codable extensions, here's one instance where you could just make it `diagnosticsEvent.prettyPrintedJSON` or `diagnosticsEvent.encodedJSON`

Reply:
thanks! much better :)